src/common/cd-special-req-component.vue

```diff
@@ -0,0 +1,67 @@
+<template>
+<div>
+  <select class="select-box form-control" v-model="specialReqSelect" @blur="onBlur" @focus="onFocus">
+    <option value="" selected data-default disabled>{{ $t('Select an option') }}</option>
+    <option value="no">{{ $t('No') }}</option>
+    <option value="yes">{{ $t('Yes') }}</option>
+  </select>
+  <input v-show="selectYes" class="form-control" v-model="specialReqInput" :placeholder="$t('Type here')" @blur="onBlur" @focus="onFocus"/>
+</div> 
+</template>
+
+<script>
+  export default {
+    name: 'cd-special-req-component',
+    props: ['value'],
+    data() {
+      return {
+        specialReqSelect: '',
+        specialReqInput: '',
+      };
+    },
+    methods: {
+      onBlur() {
+        this.blurTimeout = window.setTimeout(() => {
+          this.$emit('blur');
+        }, 50);
+      },
+      onFocus() {
+        window.clearTimeout(this.blurTimeout);
+      },
+    },
+    computed: {
+      selectYes() {
+        return this.specialReqSelect === 'yes';
+      },
+      specialRequirement: {
+        get() {
+          if (this.specialReqSelect === 'yes') {
+            return this.specialReqInput;
+          }
+          return this.specialReqSelect;
+        },
+        set(specialReqVal) {
+          if (specialReqVal === 'yes') {
+            this.specialReqSelect = 'yes';
+            this.specialReqInput = specialReqVal;
```

Comment:
Why presetting the value of specialReqInput with "yes" ?

Reply:
yeah sorry i did change that if statement so it's not really correct any longer. This is for loading the data in say when you are coming back to edit etc. so really, the if statement should be testing whether specialReqVal even exists and then setting the corresponding values to whatever they are. does that make sense?

Reply:
Yep